app/lib/backend_api_logic/routing_policy.rb

```diff
@@ -73,7 +73,15 @@ def as_json
         def replace_path
           return {} if config_path.blank?
 
-          { replace_path: "{{uri | remove_first: '#{config_path.path}'}}" }
+          { replace_path: "{{uri | remove_first: '#{liquid_path}'}}" }
+        end
+
+        private
+
+        # The replace_path rule is a liquid template. We need to escape some special characters,
+        # otherwise they won't match and won't get removed from the backend uri
+        def liquid_path
+          config_path.path.gsub(/([%+-])/, '\1\1')
```

Comment:
Can be a Lua utility:

```
# frozen_string_literal: true
class LuaUtility
  MAGIC_CHARACTERS = %w@( ) . % + - * ? [ ^  $ @
  ESCAPE_CHARACTER = "%"

  def self.escape(string)
    string.gsub(/./) do |char| 
      MAGIC_CHARACTERS.include?(char) ? "%#{char}" : char
  end
end
```

Reply:
I hadn't seen this. I will rename the module. `LuaUtility` looks nice.